module/min/mingl/src/web/model/obj.rs

```diff
@@ -0,0 +1,198 @@
+/// Internal namespace.
+mod private
+{
+  use crate::*;
+  use std::{ collections::HashSet, fmt::Display };
+  use tobj::{ Model, Material };
+  use web::model::ForBrowser;
+  use model::obj;
+
+  impl< 'model, 'mtl > Display for ForBrowser< obj::ReportObjModel< 'model, 'mtl > >
+  {
+    fn fmt( &self, f: &mut std::fmt::Formatter< '_ > ) -> std::fmt::Result 
+    {
+      fn format_vec3( v : [ f32; 3 ] ) -> String
+      {
+        format!
+        (
+          "( {}, {}, {} )", 
+          v[ 0 ], 
+          v[ 1 ], 
+          v[ 2 ]
+        )
+      }
+
+      fn format_set< V : Display >( set : &HashSet< V > ) -> String
+      {
+        let res = set
+        .iter()
+        .map( | v | v.to_string() )
+        .collect::< Vec< _ > >()
+        .join( ", " ); 
+        format!( "{{ {} }}", res)
+      }
+
+      let box_min = format_vec3( self.report.bounding_box.min.into() );
+      let box_max = format_vec3( self.report.bounding_box.max.into() );
+      let sphere_center = format_vec3( self.report.bounding_sphere.center.into() );
+      let arities_set = format_set( &self.report.num_of_arities );
+
+      write!
+      ( 
+        f, 
+        "\
+        === Model Report ===\n\
+        Name: {ModelName}\n\
+        Memory: {Memory:.2} KB\n\
+        Geometry Statistics:\n\
+        \x20 • Vertices:       {Vertices}\n\
+        \x20 • Normals:        {Normals}\n\
+        \x20 • TexCoords:      {TexCoords}\n\
+        \x20 • Vertex colors:  {VertexColors}\n\
+        \x20 • Faces:          {Faces}\n\
+        \x20 • Arities:        {Arities}\n\
+        -----------------------------------\n\
+        \x20 • Indices:        {Indices}\n\
+        \x20 • Texcoords ind.: {Tx_Indicies}\n\
+        \x20 • Normals ind.:   {N_Indicies}\n\
+        Bounding Volume:\n\
+        \x20 • Box:\n\
+        \x20    Min: {BoxMin} \n\
+        \x20    Max: {BoxMax} \n\
+        \x20 • Sphere: \n\
+        \x20    Center: {Center} \n\
+        \x20    Radius: {Radius}\n\
+        ",
+        ModelName = self.report.name,

```

Comment:
Variable names in Rust use snake_case, when types use PascalCase. Are you use clippy?

Reply:
It's not a variable name, it is a name for a pattern inside of the macro, just to be able to see the position the property goes to inside the string